File: 백준/Bronze/1157. 단어 공부/단어 공부.java
import java.io.*;
import java.util.*;

public class Main {
	public static void main(String[] args) throws IOException {
		Scanner sc = new Scanner(System.in);
		String input = sc.next();
		sc.close();

		String Uinput = input.toUpperCase();
		int[] index = new int['Z' - 'A' + 1];

		for (int i = 0; i < index.length; i++) {
			index[i] = 0;
		}

		int max = 0;
		int outIndex = 0;
		int count = 0;

		for (int i = 0; i < Uinput.length(); i++) {
			index[Uinput.charAt(i) - 'A'] += 1;
		}

		for (int i = 0; i < index.length; i++) {
			if (index[i] > max) {
				max = index[i];
				outIndex = i;
			}
		}

		for (int i = 0; i < index.length; i++) {
			if (index[i] == max) {
				count++;
			}
		}

		if (count > 1) {
			System.out.println("?");
		} else {
			System.out.println((char) (outIndex + 'A'));
		}

	}
}

File: 백준/Bronze/1157. 단어 공부/단어 공부.py
word = input()
UWord = word.upper()
Index = [0] * 26
max = 0
outIndex = 0

for alphabet in UWord:
    Index[ord(alphabet) - ord("A")] += 1

for i in range(len(Index)):
    if Index[i] > max:
        max = Index[i]
        outIndex = i + ord("A")

count = 0

for i in Index:
    if max == i:
        count += 1

if count > 1:
    print("?")
else:
    print(chr(outIndex))